### PR TITLE
support other Azure cloud environments

### PR DESCRIPTION
**Pull request by staebler**
Use the Azure cloud environment specified by the `status.platformStatus.azure.cloudName` field of the infrastructure.config.openshift.io resource.

See openshift/enhancements#321

### CODE REVIEW

**Comment by staebler** [comment]
/hold

This is currently vendoring an openshift/api fork until openshift/api#650 merges.

File: pkg/azure/actuator.go [hunk 1]
```diff
@@ -64,9 +64,11 @@ func NewActuator(c client.Client) (*Actuator, error) {
 
 	client := newClientWrapper(c)
 	return &Actuator{
-		client:                       client,
-		codec:                        codec,
-		credentialMinterBuilder:      NewAzureCredentialsMinter,
+		client: client,
+		codec:  codec,
+		credentialMinterBuilder: func(logger log.FieldLogger, clientID, clientSecret, tenantID, subscriptionID string) (*AzureCredentialsMinter, error) {
```
Comment (joelddiaz) [inline pkg/azure/actuator.go:69]: why introduce the thin wrapper around NewAzureCredentialsMinter()?

Comment (staebler) [reply pkg/azure/actuator.go:69]: `NewAzureCredentialsMinter` now takes a `cloudName` parameter. I can either have a wrapper here that closes over the `cloudName` variable, or I can add the cloud name as a field to the actuator. I chose the former.

**Comment by joelddiaz** [comment]
lgtm once this starts vendoring openshift/api

**Comment by staebler** [comment]
c6ad0bf1...f66835ed
* Bumped to the latest master version of openshift/api to include the changes from openshift/api#650.
* Rebased onto latest master

/hold cancel

**Pull request merged**